nemo_skills/inference/score_rm.py

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
I think it's better if we implement it in a way that is similar to generate.py. While your current way allows to keep the code minimal, it's likely not going to work without significant adaption for other server types (vllm / trtllm) anyway and also not going to work for api models, e.g. https://build.nvidia.com/nvidia/nemotron-4-340b-reward

So I would suggest to copy-paste most of the generate.py and then create a new server/reward_model.py which will be similar to server/model.py and provide different clients for all the server types we support.

At some point in the future we will refactor common parts of all scripts into some shared logic, but for now keeping things duplicated is fine I think

Reply:
We also most likely need to have same prompt formatting here anyway, since RM might be trained with particular chat tokens in the input, so we should also be supporting that in the same way as generate.py does